com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/pipeline/ModelUtil.java

```diff
@@ -361,79 +363,194 @@ public static ArrayList<String> loadMaterialNames(Scene scene) {
-        {
-            // if we only need 16-bit indices, use this primarily. Less data to upload to GPU and ES2.0 core functionality.
-            indices_format = Rig.IndexBufferFormat.INDEXBUFFER_FORMAT_16;
-            indices_bytes = ByteBuffer.allocateDirect(mesh_index_list.size() * 2);
-            indices_bytes.order(ByteOrder.LITTLE_ENDIAN).asShortBuffer();
-            for (int i = 0; i < mesh_index_list.size();) {
-                indices_bytes.putShort(mesh_index_list.get(i++).shortValue());
+    private static void copyIntArray(int[] src, int srcIndex, int[] dst, int dstIndex, int num_components) {
+        for (int i = 0; i < num_components; ++i) {
+            dst[dstIndex*num_components+i] = src[srcIndex*num_components+i];
+        }
+    }
+
+    private static void copyVertex(ModelImporter.Mesh inMesh, int inIndex, ModelImporter.Mesh outMesh, int outIndex) {
+        if (inMesh.positions != null) {
+            copyFloatArray(inMesh.positions, inIndex, outMesh.positions, outIndex, 3);
+        }
+        if (inMesh.normals != null) {
+            copyFloatArray(inMesh.normals, inIndex, outMesh.normals, outIndex, 3);
+        }
+        if (inMesh.tangents != null) {
+            copyFloatArray(inMesh.tangents, inIndex, outMesh.tangents, outIndex, 3);
+        }
+        if (inMesh.colors != null) {
+            copyFloatArray(inMesh.colors, inIndex, outMesh.colors, outIndex, 4);
+        }
+        if (inMesh.weights != null) {
+            copyFloatArray(inMesh.weights, inIndex, outMesh.weights, outIndex, 4);
+        }
+        if (inMesh.bones != null) {
+            copyIntArray(inMesh.bones, inIndex, outMesh.bones, outIndex, 4);
+        }
+        if (inMesh.texCoords0 != null) {
+            copyFloatArray(inMesh.texCoords0, inIndex, outMesh.texCoords0, outIndex, inMesh.texCoords0NumComponents);
+        }
+        if (inMesh.texCoords1 != null) {
+            copyFloatArray(inMesh.texCoords1, inIndex, outMesh.texCoords1, outIndex, inMesh.texCoords1NumComponents);
+        }
+    }
+
+    private static void splitMesh(ModelImporter.Mesh inMesh, List<ModelImporter.Mesh> outMeshes) {
+        int triangleCount = inMesh.indexCount / 3;
+        int vertexCount = inMesh.vertexCount;
+
+        int vcount = 0;
+        ModelImporter.Mesh newMesh = null;
+        HashMap<Integer, Integer> oldToNewIndex = null;
+        ArrayList<Integer> newIndices = null;
+
+        for (int i = 0; i < triangleCount; ++i) {
+
+            if (newMesh == null) {
+                oldToNewIndex = new HashMap<Integer, Integer>();
+                newIndices = new ArrayList<Integer>();
+
+                newMesh = new Mesh();
+                newMesh.material = inMesh.material;
+                newMesh.name = String.format("%s_%d", inMesh.name, outMeshes.size());
+
+                newMesh.texCoords0NumComponents = inMesh.texCoords0NumComponents;
+                newMesh.texCoords1NumComponents = inMesh.texCoords1NumComponents;
```

Comment:
The mechanic is straightforward.
For each new mesh we copy some values, and allocate arrays of max size.